tests/mocks/core/services/request/RequestMock.php

```diff
@@ -0,0 +1,211 @@
+<?php
+
+namespace EventEspresso\tests\mocks\core\services\request;
+
+use EventEspresso\core\domain\services\contexts\RequestTypeContextCheckerInterface;
+use EventEspresso\core\services\request\Request;
+
+/**
+ * Class RequestMock
+ * Description
+ *
+ * @package EventEspresso\tests\mocks\core\services\request
+ * @author  Brent Christensen
+ * @since   $VID:$
+ */
+class RequestMock extends Request
+{
+
+    /**
+     * $_GET parameters
+     *
+     * @var array $get
+     */
+    public $get;
+
+    /**
+     * $_POST parameters
+     *
+     * @var array $post
+     */
+    public $post;
+
+    /**
+     * $_COOKIE parameters
+     *
+     * @var array $cookie
+     */
+    public $cookie;
+
+    /**
+     * $_SERVER parameters
+     *
+     * @var array $server
+     */
+    public $server;
+
+    /**
+     * $_REQUEST parameters
+     *
+     * @var array $request
+     */
+    public $request;
+
+    /**
+     * @var RequestTypeContextCheckerInterface
+     */
+    public $request_type;
+
+    /**
+     * IP address for request
+     *
+     * @var string $ip_address
+     */
+    public $ip_address;
+
+    /**
+     * @var string $user_agent
+     */
+    public $user_agent;
+
+    /**
+     * true if current user appears to be some kind of bot
+     *
+     * @var bool $is_bot
+     */
+    public $is_bot;
+
+
+    public function __construct(array $get, array $post, array $cookie, array $server, $ip_address = '0.0.0.0')
+    {
+        $this->get = $get;
+        $this->post = $post;
+        $this->cookie = $cookie;
+        $this->server = $server;
+        $this->request = array_merge($this->get, $this->post);
```

Comment:
Is it necessary to set _all_ the above properties (other than ip_address) here?  Won't that be taken care of by `parent::__construct`?

Reply:
no because I had to override the actual properties in this class because the others are private and we don't want to change those because we don't want ppl making child classes and messing with the Request's internals.
I wouldn't do stuff like this for a regular class, but for a mock I don't really care as long as it accomplishes its goal